ompi/mpi/c/wtime.c

```diff
@@ -65,36 +58,15 @@ double MPI_Wtime(void)
 
     SPC_RECORD(OMPI_SPC_WTIME, 1);
 
-    /*
-     * See https://github.com/open-mpi/ompi/issues/3003 to find out
-     * what's happening here.
-     */
-#if 0
-#if OPAL_TIMER_CYCLE_NATIVE
-    wtime = ((double) opal_timer_base_get_cycles()) / opal_timer_base_get_freq();
-#elif OPAL_TIMER_USEC_NATIVE
-    wtime = ((double) opal_timer_base_get_usec()) / 1000000.0;
-#endif
-#else
-#if defined(__linux__) && OPAL_HAVE_CLOCK_GETTIME
+    // We intentionally don't use the OPAL timer framework here.  See
+    // https://github.com/open-mpi/ompi/issues/3003 for more details.
     struct timespec tp;
-    (void) clock_gettime(CLOCK_MONOTONIC, &tp);
-    if( OPAL_UNLIKELY(0 == ompi_wtime_time_origin.tv_sec) ) {
+    (void) opal_clock_gettime(&tp);
+    if (OPAL_UNLIKELY(0 == ompi_wtime_time_origin.tv_sec)) {
```

Comment:
Do you remember why we bother doing this work?

Reply:
I do not.  I seem to remember @bosilca having opinions about this (i.e., that we return timestamps relative to the first time Wtime is invoked).  George: do you remember?

Reply:
I talked to @bosilca about this verbally today.  His recollection is that the tools community asked for this, so that Wtime values would be smaller and therefore potentially compressible (i.e., have multiple bytes that are zero).